Make DNSSEC icon conditional in Queries Log

sharkboots75 opened the pull request:
---

**What does this PR aim to accomplish?:**

Avoid emitting the DNSSEC icon in Domain column if DNSSEC is not being used/tracked

**How does this PR accomplish the above?:**

Check for (dnssec.color !== "") which is the same convention used elsewhere in the script

![image](https://github.com/user-attachments/assets/26746d85-48e3-4e52-b13e-eb13899535db)
---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

sharkboots75 committed:
Make DNSSEC icon conditional in Queries Log

Avoid emitting the DNSSEC icon in Domain column if DNSSEC is not being used/tracked

Signed-off-by: sharkboots75 <sharkboots75@gmail.com>

rdwebdesign commented:
This is what we had before this commit https://github.com/pi-hole/web/commit/8f57898497d99245cdceea28a9d471514d177a34. 
Lines containing icons are not aligned with lines without them:

![image](https://github.com/user-attachments/assets/c1686728-a1ab-4bf6-81ab-3a9b7d1bb380)

sharkboots75 commented:
Oh I see this was done intentionally. Personally I think left aligned is the overall better option. Is it worth checking if we can render based on the global setting of DNSSEC rather than the data returned for each row?

(Noting also that long urls wrap around that icon, at least on all browsers I've tried)

rdwebdesign commented:
I agree we need to find a better solution for the icon alignment.

When we introduced this icon the initial proposal didn't align the icons at all, so I suggested [2 options (left and right aligned)](https://github.com/pi-hole/web/pull/3185#discussion_r1890857141).
We eventually decided to use the left aligned solution, but can be changed.

> (Noting also that long urls wrap around that icon, at least on all browsers I've tried)

This can be fixed using CSS. A quick solution would be to add `padding-left` to the table cell and `position: absolute` to the icon, plus `top:` and `left:` adjustments _(this will create a space on the left for the icon and fix the ugly misalignment when long domains wrap)_.
![image](https://github.com/user-attachments/assets/ad6e2d6e-b8bd-4c11-9191-9752f0d891cc)


Maybe we can use a different CSS, or something completely different for the icon.

We can leave your PR opened as a discussion while we think about solutions.

sharkboots75 committed:
Update queries.js

Added check for whether config.dns.dnssec is true so icons can be selectively rendered

Signed-off-by: sharkboots75 <sharkboots75@gmail.com>

sharkboots75 committed:
Update pi-hole.css

DNSSEC icon alignment

Signed-off-by: sharkboots75 <sharkboots75@gmail.com>

sharkboots75 commented:
Made some further changes that makes it work the way I'd like it to. 

Just a suggestion and I've only done adhoc client testing (FF, Chrome, Edge, Safari) but it's working everywhere I can check.

DNSSEC on:
![dnssec_on2](https://github.com/user-attachments/assets/fd7dbd6b-6951-427d-93a6-8eb4e6111e8b)

DNSSEC off:
![image](https://github.com/user-attachments/assets/fddcd0f4-2d97-4856-a109-29a998d6be7d)

yubiuser commented:
@sharkboots75 Tests are failing.

___

How does it look like with DNSSEC on, but lines with and without DNSSEC info? Do they still align?

yubiuser commented on scripts/js/queries.js:
```diff
@@ -26,6 +26,17 @@ var filters = [
   "reply",
   "dnssec",
 ];
+var doDnssec = false;
```
```suggestion
var doDNSSEC = false;
```

Only stylistic, but maybe we should keep DNSSEC all uppercase?

yubiuser commented on scripts/js/queries.js:
```diff
@@ -612,14 +626,17 @@ $(function () {
-        '" title="DNSSEC: ' +
-        dnssec.text +
-        '"></i>';
+      if (doDnssec !== false) {
```
```suggestion
      if (doDnssec === true) {
```

Double negation is harder to parse

sharkboots75 replied:
Yeah it is

sharkboots75 commented:
> How does it look like with DNSSEC on, but lines with and without DNSSEC info? Do they still align?

Yes, urls always align to the other urls since doDnssec setting is global (not per-row) 
![image](https://github.com/user-attachments/assets/ef9e9c5c-dbdf-4a7e-97a7-17ccae369b19)

sharkboots75 committed:
Update queries.js

Signed-off-by: sharkboots75 <sharkboots75@gmail.com>

sharkboots75 committed:
Update pi-hole.css

Signed-off-by: sharkboots75 <sharkboots75@gmail.com>

sharkboots75 committed:
Update queries.js

Changed case on var doDNSSEC

Signed-off-by: sharkboots75 <sharkboots75@gmail.com>

yubiuser commented:
Tests complaining

sharkboots75 committed:
Update queries.js

realigned to current development branch updates

Signed-off-by: sharkboots75 <sharkboots75@gmail.com>
